### PR TITLE
[MAINT] update compat to remove unusable shims e.g. py26

### PR DESCRIPTION
**Pull request by jbrockmendel**
After this, compat.scipy.NumpyVersion is never used, may be removable in a follow-up.

### CODE REVIEW

**Comment by josef-pkt** [comment]
LGTM

**Comment by bashtage** [comment]
A couple more that I think are not relevant any more:


https://github.com/statsmodels/statsmodels/blob/f8b8015d66288dad5ca717e83c1d7fd74cd70090/statsmodels/tsa/tests/test_arima_process.py#L245

RangeIndex is in Pandas 0.18+

https://github.com/statsmodels/statsmodels/blob/d3e7e53269894a1b91fe537578c5fd6bc94a1785/statsmodels/tsa/tests/test_tsa_indexes.py#L21

**Comment by jbrockmendel** [comment]
Should we get rid of NumpyVersion while we’re at it?

**Comment by bashtage** [comment]
> Should we get rid of NumpyVersion while we’re at it?

If it is completely unused then let it go.

**Comment by jbrockmendel** [comment]
Turns out we can get NumpyVersion from numpy.lib since 1.9.0.  Also some of the pandas shims are no longer needed.  Updated.

File: statsmodels/compat/pandas.py [hunk 1]
```diff
@@ -2,47 +2,21 @@
 
 from distutils.version import LooseVersion
 
-import numpy as np
 import pandas
+from pandas import RangeIndex, Float64Index  # noqa:F401
```
Comment (bashtage) [inline statsmodels/compat/pandas.py]: Ideally, anything that not needed from compat should be directly imported from the package that provides it.

File: statsmodels/compat/pandas.py [hunk 2]
```diff
@@ -2,47 +2,21 @@
...
-    def sort_values(df, *args, **kwargs):
-        # always set inplace with 'False' as default
-        kwargs.setdefault('inplace', False)
-        return df.sort(*args, **kwargs)
 
-try:
-    from pandas import RangeIndex
-except ImportError:
-    RangeIndex = tuple()
+def sort_values(df, *args, **kwargs):
```
Comment (bashtage) [inline statsmodels/compat/pandas.py]: Same here -- we shouldn't be calling  compat.sort_values (df) but df.sort_values()

**Comment by josef-pkt** [comment]
> has insisted on keeping code that is not used internally on the theory that it might be used externally.

That refers to different type of code, i.e. you wanted delete functions that are only user facing and we don't use them internally.
I don't have an opinion about NumpyVersion because I haven't looked at it in a long time. Using NumpyVersion was the old standard way for numpy, scipy, ... version check, and was guaranteed to work. (i.e. when a numpy release didn't satisfy it then there was an outcry and it was immediately changed back.)
If we don't need it anymore, then I won't miss it either.

File: statsmodels/compat/pandas.py (hunk 2 above)
Comment (jbrockmendel) [reply statsmodels/compat/pandas.py]: I agree, prefer to do this in a subsequent pass as the diff would touch a bunch of other files.

**Comment by jbrockmendel** [comment]
@josef-pkt good to go

**Comment by josef-pkt** [comment]
@jbrockmendel @bashtage Thank you

**Pull request merged**